src/main/java/com/solace/samples/java/patterns/GuaranteedProcessor.java

```diff
@@ -147,7 +147,7 @@ public static void main(String... args) throws InterruptedException, IOException
             if (inboundTopic.contains("/pers/pub/")) {  // simple validation of topic
                 // how to "process" the incoming message? maybe do a DB lookup? add some additional properties? or change the payload?
                 OutboundMessageBuilder messageBuilder = messagingService.messageBuilder();
-                messageBuilder.withApplicationMessageId(inboundMsg.getApplicationMessageId());  // set the new message ID to the same as this one
+                messageBuilder.withProperty(SolaceProperties.MessageProperties.APPLICATION_MESSAGE_ID, inboundMsg.getProperty(SolaceProperties.MessageProperties.APPLICATION_MESSAGE_ID));// set the new message ID to the same as this one
```

Comment:
Curious to know - where does it say in the docs that getApplicationMessageId is deprecated? 
Looking at the Java reference docs, I can still see this method documented 
https://docs.solace.com/API-Developer-Online-Ref-Documentation/pubsubplus-java/com/solace/messaging/Message.html#getApplicationMessageId--

Reply:
ohh.. Sorry, i referred to the wrong api, it should be the OutboundMessageBuilder.withApplicationMessageId. This is the one that is deprecated. The InboudMessage.getApplicationMessageId is fine.
I had fixed the code properly but wrote the wrong description for the PR.

Reply:
ah okay makes sense, but you are using the `getProperty` method on the inbound message to get the message ID `inboundMsg.getProperty(SolaceProperties.MessageProperties.APPLICATION_MESSAGE_ID)` instead of `inboundMsg.getApplicationMessageId()` - is there a reason for this? do these two method return the same thing?

Reply:
To answer your question : yes both the methods return the same thing. But to make it simpler, I reverted to the original code of `inboundMsg.getApplicationMessageId()`. So that should clear things up.
Also, @aaron-613 is right in saying that the api has a bug in it. To make it a bit more convenient for us, I followed what the JCSMP version is doing and did a small check to see if the`inboundMsg.getApplicationMessageId()` has a value and if yes then setting it back on the OutboundMessageBuilder.

Hope this helps.